packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -1039,15 +1163,20 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
         controller: _primaryScrollController,
         child: new Material(
           color: widget.backgroundColor ?? themeData.scaffoldBackgroundColor,
-          child: new CustomMultiChildLayout(
-            children: children,
-            delegate: new _ScaffoldLayout(
-              statusBarHeight: mediaQuery.padding.top,
-              bottomViewInset: widget.resizeToAvoidBottomPadding ? mediaQuery.viewInsets.bottom : 0.0,
-              endPadding: endPadding,
-              textDirection: textDirection,
-            ),
-          ),
+          child: new AnimatedBuilder(animation: _fabMoveController, builder: (BuildContext context, _) {
+            return new CustomMultiChildLayout(
+              children: children,
+              delegate: new _ScaffoldLayout(
+                statusBarHeight: mediaQuery.padding.top,
+                bottomViewInset: widget.resizeToAvoidBottomPadding ? mediaQuery.viewInsets.bottom : 0.0,
+                endPadding: endPadding,
+                textDirection: textDirection,
+                previousFabPosition: _previousFabPosition,
+                currentFabPosition: _fabPosition,
+                fabMoveProgress: new CurveTween(curve: Curves.ease).evaluate(_fabMoveController),
```

Comment:
rather than create this tween every frame and the MaterialArcTween every frame as well, we should probably create the chain of Animations when the target position changes, then just pass in the current value of the animation each frame.

Reply:
I've refactored as much of the tween creation as possible.

I still need to create the exact offset tween in the scaffold layout delegate because the positions still need to respond to changes in, for example, the bottom sheet or snackbar during the animation.

Reply:
Tweens are mutable, if that helps.